test/check_sharding_data_test.lua

```diff
@@ -88,6 +88,47 @@ function g.test_sharding_key_ok()
 end
 
 
+function g.test_sharding_func_ok()
+    local space = table.deepcopy(test_space)
+    space.indexes = {primary_index}
+    local schema = {
+        spaces = {
+            space = space
+        }
+    }
+
+    -- strcrc32
+    g.space.sharding_func = 'strcrc32'
+    local res, err = ddl.check_schema(schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- vshard.router.bucket_id_strcrc32()
+    g.space.sharding_func = 'vshard.router.bucket_id_strcrc32()'
+    local res, err = ddl.check_schema(schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- mpcrc32
+    g.space.sharding_func = 'mpcrc32'
+    local res, err = ddl.check_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- vshard.router.bucket_id_mpcrc32()
+    g.space.sharding_func = 'vshard.router.bucket_id_mpcrc32()'
+    local res, err = ddl.check_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
+
+    -- sharding_func defined by user
+    g.space.sharding_func = {body = 'function(key) return <...> end'}
+    local res, err = ddl.check_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(res, true)
```

Comment:
Testing `check_schema()` is good but it is not enough. I propose to test `set_schema()` too.

Reply:
Please have a look at `set_sharding_func_test.lua`, there are tests with `set_schema()`